ui_test/src/lib.rs

```diff
@@ -53,12 +68,23 @@ pub enum OutputConflictHandling {
 
 pub type Filter = Vec<(Regex, &'static str)>;
 
-pub fn run_tests(config: Config) -> Result<()> {
+pub fn run_tests(mut config: Config) -> Result<()> {
     eprintln!("   Compiler flags: {:?}", config.args);
 
     // Get the triple with which to run the tests
     let target = config.target.clone().unwrap_or_else(|| config.get_host());
 
+    let dependencies = build_dependencies(&config)?;
+    for (name, dependency) in dependencies.dependencies {
+        config.args.push("--extern".into());
+        config.args.push(format!("{name}={}", dependency.display()).into());
```

Comment:
This still uses `display` though, and not for actually displaying the filename. :/ The method is called `display` for a reason.
We don't have `format!` on `OsString` unfortunately, but we do have `push`, so concatenation should still be possible.